### PR TITLE
2.0 preview

### PR DESCRIPTION
**Pull request by e111077**
DO NOT MERGE

Major changes:
Promise polyfill moved to dev dependencies

### CODE REVIEW

File: app-indexeddb-mirror/app-indexeddb-mirror-worker.js [hunk 1]
```diff
@@ -189,6 +200,10 @@
      * transaction, or rejects if an unsupported method is attempted.
      */
     transaction: function(method, key, value) {
+      if (!self.Promise || !this.supportsIndexedDB) {
```
Comment (notwaldorf) [inline app-indexeddb-mirror/app-indexeddb-mirror-worker.js]: Wahhhhhh this repeated `if` statement blows, but I don't have a better solution

File: app-indexeddb-mirror/common-worker.html [hunk 2]
```diff
@@ -20,7 +20,11 @@
           var scripts = document.getElementsByTagName('script');
           return scripts[scripts.length - 1];
         })();
-    var BASE_URI = currentScript.ownerDocument.baseURI;
+
+    // base uri workaround for IE and Safari
+    var BASE_URI = document._currentScript ? document._currentScript.baseURI :
```
Comment (notwaldorf) [inline app-indexeddb-mirror/common-worker.html]: I think in `2.0` it's `currentScript` without the underscore, am I right? Wrong? Not sure, that's certain.

File: bower.json [hunk 3]
```diff
@@ -16,21 +16,47 @@
...
+        "platinum-sw": "polymerelements/platinum-sw#^1.3.0",
+        "promise-polyfill": "polymerlabs/promise-polyfill#^1.0.0",
+        "web-component-tester": "f27c6e0b11e476b48cdeb06e3d494ea0f132e503",
+        "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.24"
+      }
+    }
   },
-  "ignore": []
+  "ignore": [],
+  "resolutions": {
```
Comment (notwaldorf) [inline bower.json]: Don't think resolutions is needed anymore

File: app-indexeddb-mirror/app-indexeddb-mirror-worker.js (hunk 1 above)
Comment (e111077) [inline app-indexeddb-mirror/app-indexeddb-mirror-worker.js]: will remove ifs and console log the error

File: app-indexeddb-mirror/common-worker.html (hunk 2 above)
Comment (e111077) [inline app-indexeddb-mirror/common-worker.html]: making it certain...

File: bower.json (hunk 3 above)
Comment (e111077) [inline bower.json]: not any moreee

File: app-indexeddb-mirror/common-worker.html [hunk 4]
```diff
@@ -20,7 +20,18 @@
...
         })();
-    var BASE_URI = currentScript.ownerDocument.baseURI;
+
+    // fallback baseURI
+    var BASE_URI = document.currentScript ? document.currentScript.baseURI : './';
+
+    // Polymer 2 baseURI polyfill for IE and Safari
+    if (Polymer.Element && HTMLImports.importForElement) {
+      BASE_URI = HTMLImports.importForElement(document.currentScript).baseURI;
+
```
Comment (notwaldorf) [inline app-indexeddb-mirror/common-worker.html]: Delinquent new line :)

File: bower.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
   "authors": [
     "The Polymer Authors"
   ],
-  "version": "0.11.0",
```
Comment (notwaldorf) [inline bower.json]: let's check with @cdata that the version here shouldn't be `1.0.0`

File: app-indexeddb-mirror/common-worker.html (hunk 4 above)
Comment (e111077) [inline app-indexeddb-mirror/common-worker.html]: uhhhhhh....

Comment (e111077) [inline app-indexeddb-mirror/common-worker.html]: see commit msg

File: bower.json [hunk 6]
```diff
@@ -16,21 +16,43 @@
   "homepage": "",
   "private": true,
   "dependencies": {
-    "promise-polyfill": "polymerlabs/promise-polyfill#^1.0.0",
-    "polymer": "polymer/polymer#^1.2.0"
+    "polymer": "polymer/polymer#2.0.0-rc.3"
```
Comment (frankiefu) [inline bower.json]: Should this use `^` to get the latest version? Something like this: https://github.com/PolymerElements/paper-tooltip/blob/2.0-preview/bower.json#L23

Comment (e111077) [inline bower.json]: nice catch, @frankiefu

**Comment by e111077** [comment]
Tests fail in 1.9.0 due to:
https://github.com/Polymer/polymer/issues/4532

**Pull request merged**